### PR TITLE
#388 introduce multiarch images

### PR DESCRIPTION
**Pull request by lojzatran**
#### We currently use alpine/[openjdk11](https://github.com/commercetools/commercetools-project-sync/blob/5259f14189a4c95a9b9a07b6bf14909b3f5bd745/gradle-scripts/docker.gradle#L3) and this docker image does not support another arch (https://github.com/AdoptOpenJDK/openjdk-docker/issues/364). 

Therefore I switched to regular jre without alpine. It has 40 MB more than the alpine image. I tried to run the docker image and it works fine.

---

#### Currently the whole docker process is done in java (https://github.com/commercetools/commercetools-project-sync/blob/master/gradle-scripts/docker.gradle) and it does not support buildx command.

I tried to use the java plugin, but it failed on a weird login error: https://github.com/commercetools/commercetools-project-sync/runs/7769288829?check_suite_focus=true#step:5:180. In the end I had to replace the plugin with github actions. [Build is working](https://github.com/commercetools/commercetools-project-sync/runs/7784518717?check_suite_focus=true), the image is multiplatform now. I pulled it and ran it and it is working fine.
Screenshot 2022-08-11 at 12 06 31

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM adoptopenjdk/openjdk11:jre
```
Comment (praveenkumarct) [inline Dockerfile:1]: Adoptopenjdk seems like depreciated.
https://github.com/AdoptOpenJDK/openjdk-docker#-deprecation-notice-

Also looks like the AdoptOpenJDK moved to [Eclipse Adoptium](https://projects.eclipse.org/projects/adoptium):
 https://blog.adoptopenjdk.net/2021/08/goodbye-adoptopenjdk-hello-adoptium/

May be explore a bit on this and completely move out of adoptopenjdk?
 
Examples of some PR's to migrate to Eclipse temurin in this issue:
https://github.com/docker-library/openjdk/issues/505


If you think, we need Evaluation or needs more time to understand, please create a ticket.

**Review by praveenkumarct — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (lojzatran) [reply Dockerfile:1]: Good catch 🐱 However, it is not in scope of my task. Thus I created follow-up ticket for this: https://jira.commercetools.com/browse/CTPI-698

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -83,13 +83,18 @@ It is recommended to set it in your IDE auto formatting settings for this projec
...
+### Build a docker image
+
+To build a docker image, you need to install docker. [Dockerfile](../Dockerfile) is provided in the root folder.
+Before building a docker image, you need to compile java classes:
+```bash
+./gradlew setLibraryVersion assemble
+```
+Then you can build the docker image with the following command:
+```bash
+docker buildx build  --platform linux/amd64 --tag commercetools/commercetools-project-sync:<version> . --load
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply docs/CONTRIBUTING.md]: https://github.com/commercetools/commercetools-project-sync/pull/393/commits/4cfb4cf7b4155c4340cd795bf044e832c32ef502

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -20,7 +20,21 @@ jobs:
...
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and push extension module
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply .github/workflows/cd.yml]: https://github.com/commercetools/commercetools-project-sync/pull/393/commits/069d2511a8b87d5678aaade55a7c1eea6697a80d

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -20,7 +20,21 @@ jobs:
           java-version: 11
           distribution: 'zulu'
       - name: Build and push image to docker using gradle plugin
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply .github/workflows/cd.yml]: https://github.com/commercetools/commercetools-project-sync/pull/393/commits/069d2511a8b87d5678aaade55a7c1eea6697a80d

**Review by praveenkumarct — APPROVED** [review]

**Pull request merged**